File: csharp/high-scores/HighScores.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
    private List<int> data;

    public HighScores(List<int> list)
    {
        data = list;
    }

    public List<int> Scores()
    {
        return new List<int>(data);
    }

    public int Latest()
    {
        return data.Last();
    }

    public int PersonalBest()
    {
        return data.Max();
    }

    public List<int> PersonalTopThree()
    {
        return new List<int>(data.OrderByDescending(x => x).Take(3));
    }
}

File: java/high-scores/src/main/java/HighScores.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;
import java.util.stream.Collectors;

public final class HighScores {

    private List<Integer> scores;

    public HighScores(List<Integer> highScores) {
        scores = new ArrayList<>(highScores);
    }

    List<Integer> scores() {
        return Collections.unmodifiableList(scores);
    }

    Integer latest() {
        return scores.get(scores.size() - 1);
    }

    Integer personalBest() {

        return scores.stream()
                     .max(Integer::compare)
                     .get();
    }

    List<Integer> personalTopThree() {

        return scores.stream()
                     .sorted(Comparator.reverseOrder())
                     .limit(3)
                     .collect(Collectors.toList());
    }
}
